subsys/bluetooth/shell/bt.c

```diff
@@ -1317,6 +1420,40 @@ static int cmd_auth_passkey(const struct shell *shell,
 #define HELP_NONE "[none]"
 #define HELP_ADDR_LE "<address: XX:XX:XX:XX:XX:XX> <type: (public|random)>"
 
+SHELL_CREATE_STATIC_SUBCMD_SET(sub_bt_advertise_on_cmds)
+{
+	/* Alphabetically sorted. */
+	SHELL_CMD(discov, NULL, "Advertise connectable on", cmd_advertise_conn_discov),
+	SHELL_CMD(ndiscov, NULL, "Advertise connectable non-discoverable", cmd_advertise_conn_ndiscov),
+	SHELL_SUBCMD_SET_END /* Array terminated. */
+};
+
+SHELL_CREATE_STATIC_SUBCMD_SET(sub_bt_advertise_scan_cmds)
+{
+	/* Alphabetically sorted. */
+	SHELL_CMD(discov, NULL, "Advertise scannable on", cmd_advertise_scan_discov),
+	SHELL_CMD(ndiscov, NULL, "Advertise scannable non-discoverable", cmd_advertise_scan_ndiscov),
+	SHELL_SUBCMD_SET_END /* Array terminated. */
+};
+
+SHELL_CREATE_STATIC_SUBCMD_SET(sub_bt_advertise_nconn_cmds)
+{
+	/* Alphabetically sorted. */
+	SHELL_CMD(discov, NULL, "Advertise non-connectable on", cmd_advertise_nconn_discov),
+	SHELL_CMD(ndiscov, NULL, "Advertise non-connectable non-discoverable", cmd_advertise_nconn_ndiscov),
+	SHELL_SUBCMD_SET_END /* Array terminated. */
+};
+
+SHELL_CREATE_STATIC_SUBCMD_SET(sub_bt_advertise_cmds)
+{
+	/* Alphabetically sorted. */
+	SHELL_CMD(on, &sub_bt_advertise_on_cmds, "Advertise connectable on", cmd_advertise_conn_discov),
```

Comment:
We should definitely use use SHELL_CMD_ARG whenever possible.

Reply:
Btw, I don't think it is very useful to have the description of the command itself on subcommands but the mandatory <> and optional [] parameters, otherwise we kind have to duplicate the command description on to subcommands.

Reply:
modified it a bit. #11620 fixes printing help for 0 mandatory arguments so printing help will be supported for all stages.